### PR TITLE
Sign Up

### PR DESCRIPTION
**Pull request by brokenheart1712**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      render 'new'
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation)
+  end
+end
```
Comment (NguyenTanDuc) [inline app/controllers/users_controller.rb]: Add new line

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      <%= f.label :password, "Password: " %>
+      <%= f.password_field :password, class: 'form-control' %>
+
+      <%= f.label :password_confirmation, "Confirmation: " %>
+      <%= f.password_field :password_confirmation, class: 'form-control' %>
+
+      <%= f.submit "Create My Account!", class: "btn btn-primary" %>
+    <% end %>
+  </div>
+</div>
```
Comment (NguyenTanDuc) [inline app/views/users/new.html.erb]: Add new line

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="row">
+  <div class="col-md-3">
+    <p align="center">
+      <div style="text-align:center;">
+        <%= @user.name %>
+      </div>
+    </p>
+  </div>
+  <h3>Activities </h3>
+</div>
```
Comment (NguyenTanDuc) [inline app/views/users/show.html.erb]: Add new line

File: db/migrate/20150508041512_create_users.rb [hunk 4]
```diff
@@ -3,11 +3,12 @@ def change
...
+      t.boolean :admin
       t.string :avatar
+      t.string :password_digest
+      t.string :password_confirmation
 
       t.timestamps null: false
     end
   end
-end
+end
```
Comment (NguyenTanDuc) [inline db/migrate/20150508041512_create_users.rb]: Add new line

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  root                'static_pages#home'
+  root 'static_pages#home'
+  resources :users
```
Comment (NguyenTanDuc) [inline config/routes.rb]: Just use what do you need

**Comment by NguyenTanDuc** [comment]
OK now.

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    if @user.save
+      flash[:success] = "Welcome To The Framgia E-Learning System!"
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+  
+  def edit
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+  
+  def edit    
+  end
+  
+  def update
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    end
+  end
+  
+  def edit    
+  end
+  
+  def update    
+  end
+  
+  def destroy
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  def update    
+  end
+  
+  def destroy    
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation)
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/models/user.rb [hunk 12]
```diff
@@ -1,10 +1,18 @@
...
+                                  dependent: :destroy
   has_many :passive_relationships, class_name:  "Relationship",
                                    foreign_key: "followed_id",
                                    dependent:   :destroy
-  has_many :following, through: :active_relationships,  source: :followed
+  has_many :following, through: :active_relationships, source: :followed
   has_many :followers, through: :passive_relationships, source: :follower
+  has_many :lessons
+  has_many :activities
+  validates :name, presence: true
```
Comment (dieunb) [inline app/models/user.rb]: add new line above to slice code to many blocks

File: app/views/shared/_error_messages.html.erb [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<% if @user.errors.any? %>
+<div id="error_explanation">
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: indent

File: app/views/shared/_error_messages.html.erb [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  <div class="alert alert-danger">
+    The form contains <%= pluralize(@user.errors.count, "error") %>.
+  </div>
+  <ul>
+    <% @user.errors.full_messages.each do |msg| %>
+    <li>
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: indent

File: app/views/users/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,24 @@
+<% provide(:title, 'Sign up') %>
+<h1>Sign up</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (dieunb) [inline app/views/users/new.html.erb]: remove `()`

File: app/views/users/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,24 @@
+<% provide(:title, 'Sign up') %>
```
Comment (dieunb) [inline app/views/users/new.html.erb]: remove `()`

File: app/controllers/users_controller.rb [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+  
+  private
+
+  def user_params
+    params.require :user.permit(:name, :email, :password, :password_confirmation)
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/views/shared/_error_messages.html.erb [hunk 18]
```diff
@@ -0,0 +1,14 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: remove `()`

File: config/routes.rb (hunk 5 above)
Comment (dieunb) [inline config/routes.rb]: Filter routes by using `only` or `except`

**Pull request merged**